templates/energyplus/templates/ref-sys.pxt

```diff
@@ -121,3 +126,122 @@ Refrigeration:Condenser:AirCooled,
     22.2;                    !- Maximum Value of x
 <% end %>
 
+#Afshin Faramarzi and Behzad Rizi, 11/11/2024 - added EnergyManagementSystem to the model to control saturated condensing temperature 
+!-   ===========  ALL OBJECTS IN CLASS: ENERGYMANAGEMENTSYSTEM:ACTUATOR ===========
+
+<% if (ref_system_name == "MT_System") %>
+<% if (condenser_type == "AirCooled") %>
+
+EnergyManagementSystem:Sensor,
+Tamb,                    !- Name
+*,                       !- Output:Variable or Output:Meter Index Key Name
+Site Outdoor Air Drybulb Temperature;  !- Output:Variable or Output:Meter Name
+
+EnergyManagementSystem:Actuator,
+    CondensingTempOverrideMT,!- Name
+    MT_System,               !- Actuated Component Unique Name
+    Refrigeration:System,    !- Actuated Component Type
+    Minimum Condensing Temperature;  !- Actuated Component Control Type
+	
+EnergyManagementSystem:OutputVariable,
+MT Ref Sys Cond Temp,    !- Name
+CondensingTempOverrideMT,!- EMS Variable Name
+Averaged,                !- Type of Data in Variable
+SystemTimestep,          !- Update Frequency
+CondTempControlMT,       !- EMS Program or Subroutine Name
+C;                       !- Units	
+	
+EnergyManagementSystem:Program,	
+CondTempControlMT,       !- Name
+SET CondensingTempOverrideMT = Tamb + <%= cond_sct_delta %>;  !- Program Line 1
+
+EnergyManagementSystem:ProgramCallingManager,
+    Ref Sys Cond Temp Management MT,  !- Name
+    InsideHVACSystemIterationLoop,  !- EnergyPlus Model Calling Point
+    CondTempControlMT;       !- Program Name 1
+	
+<% elsif (condenser_type == "EvaporativelyCooled") %>
+
+EnergyManagementSystem:Sensor,
+Twet,                    !- Name
+*,                       !- Output:Variable or Output:Meter Index Key Name
+Site Outdoor Air Wetbulb Temperature;  !- Output:Variable or Output:Meter Name
```

Comment:
Within the branches of the IF-statement for condenser_type, there is a lot of redundancy between air-cooled and eva-cooled. Could we simplify it so that the IF-statement is only within the EnergyManagementSystem:Sensor object? For compatibility with the EMS Program, keep the sensor name unchanged. To avoid duplicate objects, also use the system name. For example, 


    EnergyManagementSystem:Sensor,
    T_sensor_<%= ref_system_name %>,                    !- Name
    *,                       !- Output:Variable or Output:Meter Index Key Name
    <% if (condenser_type == "EvaporativelyCooled") %>
    Site Outdoor Air Wetbulb Temperature;  !- Output:Variable or Output:Meter Name
    <% else %>
    Site Outdoor Air Drybulb Temperature;  !- Output:Variable or Output:Meter Name
    <% end %>

Reply:
Agreed. Revised based on your comment.